core/domain/exp_services.py

```diff
@@ -839,19 +817,17 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
 
     exploration_model = exp_models.ExplorationModel.get(
         exploration.id, strict=False)
-    if exploration_model is None:
-        exploration_model = exp_models.ExplorationModel(id=exploration.id)
-    else:
-        if exploration.version > exploration_model.version:
-            raise Exception(
-                'Unexpected error: trying to update version %s of exploration '
-                'from version %s. Please reload the page and try again.'
-                % (exploration_model.version, exploration.version))
-        elif exploration.version < exploration_model.version:
-            raise Exception(
-                'Trying to update version %s of exploration from version %s, '
-                'which is too old. Please reload the page and try again.'
-                % (exploration_model.version, exploration.version))
+
+    if exploration.version > exploration_model.version:
```

Comment:
On line 819 strict is False so the exploration model can be None, we removed this because we are doing such check before calling this method?

Reply:
Actually, exploration is passed as an argument here and its only possible if it has an associated model. We cannot directly pass exploration domain object here if that's what you are thinking since its a private function. It is called by update_exploration and that in turn calls apply_change_list. And apply_change_list will throw an error if it gets an invalid exp_id (see first line of apply_change_list)

Reply:
> exploration is passed as an argument here and its only possible if it has an associated model

I don't think this is true!

> apply_change_list will throw an error if it gets an invalid exp_id (see first line of apply_change_list)

I agree with you on this, I think  _save_exploration method was created also for using while creating exploration but we don't use it for such case anymore.

Reply:
_save_exploration is a private method and its only called by one public method(update_exploration) which checks if the exploration has a model or not. This method(_save_exploration) cannot be called directly hence we cannot put an exploration with no model

Reply:
I have already accepted this change. 

Also, if we aren't expecting any None value here then can we change line 827 with strict=True?

Reply:
Done

Reply:
I think the default value is True so we don't have to mention it. Can you please check?

Reply:
Yes the default is True. Done